charter: Add sections on membership, resignations, and removals (#248)
kubernetes/steering: charter.md

* Control and delegate access to and establish processes regarding
  project resources/assets<sup>[6](#footnote6)</sup>  

## Changes

In instances where a process is not already specified within this document,
changes to the Steering Committee charter will be considered according to the
processes set forth in the committee's [operations documentation][changes].

[changes]: /operations/changes.md

## Membership

### Elections

Every year, the Steering Committee holds a general election for open seats with
[SIG Contributor Experience][sig-contribex] owning the process.

> SIG Contributor Experience may delegate this responsibility to a subproject.

Reviewer: technically we don't have to say this. but its ok to leave it in.

Our [election policy document][general-elections] covers the details for how
this works.

Policy and operations of special elections initiated by the Steering Committee
for other reasons (as yet to be determined) may be handled by some agreed-upon
[process][changes] by the committee.

[general-elections]: /elections.md
[sig-contribex]: https://git.k8s.io/community/sig-contributor-experience

### Vacancies

In the event of a resignation, removal, or other loss of an elected Steering 
Committee member, the candidate with the next most votes from the previous 

> election will be offered the seat. This process will continue until the seat is 

Reviewer: i think this may be made a bit clearer perhaps?

`If the person offered the seat declines, we repeat this process until the seat is filled`

Reviewer 2: I continue to think that a maximum of 1 board seat should be filled this way between elections. The further down the list you go, the less likely it is that the majority of voters actually wants that person.

Reviewer 3: > The further down the list you go, the less likely it is that the majority of voters actually wants that person.

Great point. We'll consider what could be a more complete way to handle this.

Reviewer 2: There is some more discussion of this buried in comment on an old revision, but the other consideration is that you can leave the seat empty if it's not too long until the next regular election, and doing that is probably the best choice if e.g. the next election is less than a month away.

Reviewer 4: Unsurprisingly, I agree with Daniel on this topic.  The list of candidates here is self-limiting - many people don;t run because they have colleagues who are running and the max-rep rules make it an XOR for them.  As such the set of people who ran "last time" does not necessarily represent the set of people who might be most appropriate if an election were held anew.

Reviewer 5: This is not a new rule; it's in the existing election rules.  We might want to change it later, but maybe hold off on that?

Reviewer 2: It's not a new rule but it interacts poorly with new rules, so it should probably be changed now.

Reviewer 5: I'd planned to have the new Elections Subproject* take up this particular bit, which has needed improvement for some time, but we need an alternate procedure which we'd need to hash out.

* https://github.com/kubernetes/community/pull/6720

Reviewer 5: If we are not waiting:

+1 on limiting replacement-by-next-most-preferred to one per year, and then going to a special election, unless the next general election is less than three months away.

filled. 

> In case this fails to fill the seat, a special election for that position will 

Reviewer: `this` -> `this process`

be held as soon as possible. [Eligible voters](https://git.k8s.io/steering/elections.md#eligibility-for-voting) from the most recent election 
will vote in the special election (ie: eligibility will not be redetermined at 
the time of the special election). A committee member elected in a special 
election will serve out the remainder of the term for the person they are 
replacing, regardless of the length of that remainder.

### Resignation

If a committee member chooses not to continue in their role, for whatever
self-elected reason, they must notify the committee in writing. As a courtesy,

> such notifications should be given at least 30 calendar days in advance of

Reviewer: any specific reason for 30 days? (could we just do 2 weeks?)

Reviewer 6: I would advocate for a removal of any time frame. If someone has to resign with immediate effect (for family or health reasons for example) I wouldn't want them to have to feel guilt for doing so.

their departure.

The committee can choose to accept a member's resignation and carry out

> offboarding procedures at any time following the written notice.

Reviewer: does offboarding also imply ... onboarding for the replacement?

### Removal

#### No confidence

> A Steering Committee member may be removed by a two-thirds vote of no

Reviewer 2: After thinking about this overnight, I think this should require unanimity (sans target). Since it is overriding the will of the voters. Requiring unanimity will make abuse of this process drastically harder while not making it that much harder to eject someone who definitely deserves it.

Reviewer 2: The other logical choice would be to copy the 3/4ths threshold (presumably rounded up, although not stated) from the [change process](https://github.com/kubernetes/steering/blob/main/operations/changes.md). That works out to the same number for 7 seats but might make more sense for larger numbers of seats.

I do not think it should be easier to eject a board member than it is to change the rules.

Reviewer 6: I think I agree with the 3/4ths suggestion.

While in theory I would prefer unanimity (we strive for all steering decisions to be a consensus whenever possible), I think the logistics make it difficult. Some folks may wish to abstain (such as if the person being removed works for the same employer), and it's just not great practice to give a single person on the committee a veto over any decision (which would be the result of a requirement of unanimity).

Reviewer 2: 7 * 3/4 = 5.25, round up is 6. So for this board size it doesn't change the number of votes.

I agree in general, the board should be able to function with a single rogue actor. But for the specific question of ejecting someone, I personally want the rest of the board to be unified about that.

I also think the general threshold should be 50% + 1, not 2/3rds -- then 3/4ths would be a more meaningful step up.

> we strive for all steering decisions to be a consensus whenever possible

It's a good aspiration! But the rules of course need to be written assuming that it's not working and people are going to have to vote on something important to them. If that was an achievable goal, we wouldn't need rules at all. Hence why e.g. Robert's Rules assume fairly antagonistic conditions, I guess.

confidence from other Steering Committee members.

> The call for a vote will happen in a public Steering Committee meeting and a

Reviewer: `vote` -> `vote of no confidence`

GitHub issue in kubernetes/steering by a current member and must be seconded by

> another.

Reviewer: `another current member` just to round it off.

> another.

Reviewer: Do we need a mechanism to notify the community (push this news to k8s-dev / steering mailing lists? once the github issue is ready with some context for the vote of no confidence)

> There is a one week community feedback period for contributors to use to reach

Reviewer: contributors -> members of the community (may sound better?)

also "to use to reach" could just be "to reach"

> There is a one week community feedback period for contributors to use to reach

Reviewer: is -> will be?

out to Steering Committee members with their thoughts or log them on the open

> issue including non-binding votes. After that period, Steering Committee

Reviewer: `reach out to Steering Committee members with their thoughts or log them on the open issue including non-binding votes` mixes up a few things, lets separate them out?

`reach out to Steering Committee members individually or collectively using the usual mechanisms (steering public or private mailing lists, steering slack etc). They can also log their thoughts in the github issue (non-binding votes are welcome!).

Reviewer 2: I would discourage members from publishing non-binding votes in advance because it makes it harder for them to change their minds and will create the impression that the board was not sensitive to feedback recieved.

Reviewer 6: I'm concerned about the possibility of "brigading" on a particular impacted person if we encourage public comment on a sensitive issue like removing someone from a leadership position.

It's tricky, because allowing community members to share feedback is critical, and having transparency is important, but for many reasons, we also don't want it to turn into a mob attack either.

> members have 48 hours to vote on the issue.

Reviewer: have have 48 hours to vote on the issue -> must vote within 48 hours? (future tense?)

* Control and delegate access to and establish processes regarding
  project resources/assets<sup>[6](#footnote6)</sup>  

## Changes

In instances where a process is not already specified within this document,
changes to the Steering Committee charter will be considered according to the
processes set forth in the committee's [operations documentation][changes].

[changes]: /operations/changes.md

## Membership

### Elections

<!--

- Consider explicitly mentioning SIG ContribEx's Elections subproject, once it
exists.

-->

Every year, the Steering Committee holds a general election for open seats with
[SIG Contributor Experience][sig-contribex] owning the process.

Our [election policy document][general-elections] covers the details for how
this works.

In the instance that a general election is called as a result of the
[dissolution](#dissolution) of the current committee, the following rules will
apply:

* members of the current committee will not be allowed to run in the new

>   election

Reviewer 7: This is a fairly big deal. Current members are no longer allowed to run again without ever having violated the CoC or going through any judicial governance process.

In fact, 4 members of the steering committee could resign at the same time and not say why. This poison pill would stop people from running in the new election and the community would never know the reason. Did the people who didn't resign do something wrong? It wasn't CoC based. What's going on? Community members may not know.

This is not transparent (which is a k8s principle).

Reviewer 2: That bothers me too, but what can be done about it? We can't prevent people from resigning and if enough people resign at a time we'd have to have an election to replace them.
* Only run an election for the vacant seats? (with most popular winners getting the longest-duration seats)
* Run an election for everyone, but only prevent those who resigned from running?

I don't really think either of these options is great (not a huge fan of the option in the current revision, either, though)

Reviewer 5: I believe the revision Matt is asking for is to remove the part that says that nobody on the dissolved SC can run in the subsequent election.  I agree with that critique.  Among other things, it could leave us in a situation where we don't have enough candidates to reform the SC.

Reviewer: > This is not transparent (which is a k8s principle).

FWIW we are where we are today because of folks not adhering to k8s principles - @mattfarina - hope you would be up to call out that set of behaviors as well Matt.

Reviewer 3: (Still noodling on this section...)

Reviewer 7: We operate in a very diverse area around a technical project.

To illustrate diversity we can look at ideology. We have people with varying belief backgrounds including atheist, christian, Muslim, Hindu, and many others. Then there are the political ones such as Renaissance, Socialist Party, Free Voters, National Renewal, Republicans, and Democrats. I started by listing non-US political parties because we are an international crowd with contributions from so many different countries. It can be easy to miss this with a North American based steering committee (only one member outside the US and that one is still in North America).

Everyone comes together for the technology (i.e. Kubernetes). Even when they oppose each other in some other space.

This is why things like transparency and inclusiveness are so important (a value and a principle), IMHO. Including people in the technical conversations and activities of the project no matter their politics, religion, gender, or other characteristic. Transparency so that people in different environments, who sometimes can't pick up on the social aspects in one geo, can follow and make well informed decisions.

There is a time to not have transparency. CoC violation details is an easy example. But, for big things like removing a steering committee member for a non-CoC issue or the dissolution of the steering committee we need transparency. So this cross region, cross belief, cross political, cross primary language community can navigate what's going on.

That's just my 2 cents.

Reviewer 2: @dims and @mattfarina I think it would be best if in this particular place we stuck to debating how to wind things down fairly and **not** why one might want to do so. Otherwise we will overtrain on the emergency of the moment and make a policy that doesn't generalize, which is even worse for the board in the long term. The time for the discussion you are having is for later, when people are trying to wield the policy described here. (TLDR this is very close to derailing the conversation here, which has so far been very productive.)

Reviewer: @lavalamp fair advice. thanks!

Reviewer 4: I have a slight lean towards striking this line, myself. But that does not address the transparency issue at the top of this thread.  Whether anyone can re-run or not, there's nothing we can do to force them to make a statement or for that statement to be useful or even true.

We could get draconian and say something like "We ask resigning members to make a statment of intent.  If a resigning member chooses to run in the new election, they must make such a statement (100 words or less)."  But that STILL doesn't make it true or relevant or sensical.

"I resigned because I needed to buy milk, but now that I have milk I am interested in serving again."

If we do keep this rule, we should maybe be extra-clear whether this applies to ALL members or just resigning members, and that those people may (?) run in future elections.

Reviewer: @thockin it's not just about the resigning members, but also members who will NOT resign as well. The intent was one subsequent election. Think of when steering is fragmented enough to be dysfunctional and the folks resigning will save trouble for future members that fill their seats.

* a new committee will be bootstrapped with:
  * the top four candidates serving a two-year term

>   * the next three candidates serving a one-year team

Reviewer 5: ```suggestion
  * the next three candidates serving a one-year team
* the Election Officers will be selected by the Elections Subproject and confirmed by the Contributor Experience Chairs.
```

We'll need a way to pick the EOs in the absence of a functioning SC.  My suggestion is above.  I'm not attached to having ContribEx pick the EOs, but we need to have someone other than the SC approve them (the process in a normal election).

Author: agree with dims - have contribex select the EO team as they do today and then the final approvers are the leads@ list. the only thing that needs to change is the approvers.

Reviewer 5: Paris: you know as well as I do that folks don't respond on the leads@ list -- certainly we aren't going to get a majority of leads responding, if we get any at all.  So that would mean we'd need to do approval-by-lazy-consensus, which means that it would take a minimum of a week for the EOs to be approved.

Reviewer 5: Again, the threat model here is "unable to reform the SC because holding a special election is too hard".   Let's not make that threat worse.

Reviewer 2: There are multiple threat vectors here. You can't optimize for one only.

Perhaps we should go with, the election officers from the last election must run this one. And add that retroactively to the requirements for being an election officer, that you're potentially on the hook for running a special one. This isn't fair to the people who didn't sign up for that but hopefully they can forgive us.

Reviewer 2: It's somewhat poetic for the security response committee to assist in rebooting the board. That's not what they signed up to do, but that likely makes them a good check on the system.

Reviewer 5: I'm happy with any relatively small group of people who are known to be responsive.  SRC and/or COCC work for that.

Policy and operations of special elections initiated by the Steering Committee
for other reasons (as yet to be determined) may be handled by some agreed-upon
[process][changes] by the committee.

[general-elections]: /elections.md
[sig-contribex]: https://git.k8s.io/community/sig-contributor-experience

### Vacancies

In the event of a resignation, removal, or other loss of an elected Steering 

> Committee member, the candidate with the next most votes from the previous 

Reviewer 5: ```suggestion
Committee member, the next most preferred candidate from the previous 
```

Per below, I'd like to hold off on revising this section because I wanted to do it through the Elections Subproject.  However, if we are revising it, the phrase "next most votes" needs to go away, because it's not accurate.  We use preference elections.

Author: agree with you. lets not change this language right now.

election will be offered the seat. This process will continue until the seat is 
filled. 

In case this process fails to fill the seat, a special election for that position will
be held as soon as possible. [Eligible voters](https://git.k8s.io/steering/elections.md#eligibility-for-voting) from the most recent election 
will vote in the special election (ie: eligibility will not be redetermined at 
the time of the special election). A committee member elected in a special 
election will serve out the remainder of the term for the person they are 
replacing, regardless of the length of that remainder.

### Resignation

If a committee member chooses not to continue in their role, for whatever
self-elected reason, they must notify the committee in writing.

> ### Dissolution

Reviewer 2: The point of the staggered terms is so the board has continuity; both of these options would lose that. I have been on a board in the past and continuity is super super important.

I would consider adding something like:

#### Continuity

If the board is dissolved completely by any mechanism, selected members (selection described below) agree to be "continuity advisors" and attend meetings of the subsequently elected incoming board in a non-voting capacity for 6 meetings or until the incoming board dismisses them, whichever is first.

Selection. At least two and not more than three members shall be selected for this duty.
* One shall be the longest-serving outgoing board member. If multiple, the incoming board may select between them.
* The second shall be selected by the incoming board at their discretion.
* The incoming board, at their discretion, may select a third.

Reviewer 5: +1

Reviewer 3: > The point of the staggered terms is so the board has continuity; both of these options would lose that. I have been on a board in the past and continuity is super super important.

Great point. Added this language in https://github.com/kubernetes/steering/pull/248/commits/e0ffe089051dd731f33b19f1e32a19a8a25a0094.

If committee members believe that the committee is no longer able to act in
accordance with the above [mission](#mission), the committee may dissolve by a
few methods:

* Dissolution by vote
* Dissolution by loss of committee

See [elections](#elections) for special criteria in the instance of a general
election triggered by dissolution.

#### Dissolution by vote

The committee must specify a date of dissolution. Dissolution requires an

> affirmative vote of 75% of committee members, and must be unanimous when the

Reviewer 2: Rounded up? Actually I think this one could be a lower threshold, since it's costly to the board members invoking it. I would say 2/3rds round up is more appropriate. The real losers in this scenario are those running the subsequent election.

But I think this whole section is redundant given the next section, which has a lower bar.

committee has four (4) or fewer members. If the committee is dissolved, all
seats are vacated on the date specified.

#### Dissolution by loss of committee

In the event that 50% (or more) of the committee's members

> [resign](#resignation) at the same time, the entire committee will be disbanded.

Reviewer 2: I think it's critical in both of these sections to specify if the outgoing members can run in the subsequent election or not. Making that different for the two mechanisms here might give a reason for there being two mechanisms.

Reviewer 5: Need to be clear that this is 50% "rounded up", e.g. 4 out of 7

> [resign](#resignation) at the same time, the entire committee will be disbanded.

Reviewer 2: I agree "same time" is ambiguous and I'd say instead "if at any time 50% of the board's seats are vacant" and also state whether we evaluate this before or after we replace people from the prior election's results.

### Removal

#### No confidence

> A Steering Committee member may be removed by a three-fourths vote of no

Reviewer 2: It should be implicit, but I would add (rounded up) for extra clarity.

Reviewer 2: (and thank you for changing it)

> A Steering Committee member may be removed by a three-fourths vote of no

Reviewer 5: ```suggestion
A Steering Committee member may be removed by a vote of at least 75% of existing members of no
```

That makes it clearer.  Note that, at the present time, that requires all 6 of the non-removed members to unanimously vote.

Author: I think the original language blends with the doc better and we use that in other places but if others agree - lgtm

Reviewer 2: The thing that needs clarifying is that it should be rounded up IMO, and the denominator.

for style reasons the doc should pick fractions or percent and use it universally, but that's just a nit.

Reviewer 8: +1 on clarifying the rounding up (as was done in other examples above)

Reviewer 5: My change tries to clarify two things: (1) that it's 75.000% *or more*, rounded up and (2) that it's a % of the existing SC members not just the ones who show up for a meeting.  We could use a variety of different text for that, but those are the two numbers that need to be clear.

Reviewer 2: I think we are all agreeing, we just need the text updated.

These votes may proceed on agreed-upon platforms by the committee, such as:

* a pull request
* an issue
* a Steering Committee [meeting](#meetings)
* a mailing list

Unless otherwise specified by a process, the requirement for passing a vote is
a **_two-thirds supermajority of the
[fixed membership of the committee](#composition)_**.

Example:

* 7 (members) / 3 = 2.333...
* 2.333... * 2 = 4.666...
* Round up to the nearest whole number (5)
* 5 members in attendance would be required to vote during a meeting

### Abstention

For any self-elected reason, members of the committee may decide to abstain
from a vote.

Abstaining members will only be considered as contributing to quorum, in the
event that a vote is called in a meeting.

## Meetings

Steering Committee members are generally expected to attend every meeting. We
use the following guidelines to determine whether we have reached quorum and
are able to proceed with a meeting.

### Quorum

Quorum **to meet** is a **_majority of the
[fixed membership of the committee](#composition)_**.

Example:

* 7 (members) / 2 = 3.5
* Round up to the nearest whole number (4)
* 4 members in attendance would be required to meet

> Quorum **to vote in a meeting** is a **_two-thirds supermajority of the

Reviewer 5: This says that you need 50% to have a meeting, but 67% to vote in a meeting.

What's the point of a meeting where no votes can be taken? Why even have that in the rules?

Reviewer 2: It's to prevent part of the board plotting against the rest of the board :/

At least that was my understanding of similar provisions I've seen.

Reviewer 3: > What's the point of a meeting where no votes can be taken? Why even have that in the rules?

There are meetings where we do not have votes on the agenda, so in many instances, a quorum of a majority of the fixed membership of the committee is sufficient to proceed.

### Vacancies

In the event of a resignation or other loss of an elected steering committee
member, the candidate with the next most votes from the previous election will
be offered the seat.  This process will continue until the seat is filled.

In case this fails to fill the seat, a special election for that position will
be held as soon as possible. [Eligible voters][voter-eligibility] from the most
recent election will vote in the special election i.e., eligibility will not be
redetermined at the time of the special election. A committee member elected in
a special election will serve out the remainder of the term for the person they
are replacing, regardless of the length of that remainder.

[voter-eligibility]: /elections.md#eligibility-for-voting

### Resignation

If a committee member chooses not to continue in their role, for whatever
self-elected reason, they must notify the committee in writing.

### Removal

#### No confidence

A Steering Committee member may be removed by an affirmative vote of a
**_three-quarters supermajority of the
[fixed membership of the committee](#composition)_**, and must be unanimous

> when the committee has four (4) or fewer members.

Reviewer 8: Good point, I think this case it would make sense to consider this an abstention by default. Lang can be updated via the work done in https://github.com/kubernetes/steering/issues/250 ?

> when the committee has four (4) or fewer members.

Reviewer 8: I was working with the assumption that unanimous meant everyone _but_ the person being potentially removed. 

💯 agree this needs clarification so that we can clearly understand the intentions and voting requirements when there are 4 members so that it's not left to presumptions and interpretations. 👍

> when the committee has four (4) or fewer members.

Reviewer 2: I agree, thank you.

> when the committee has four (4) or fewer members.

Reviewer 8: That works!

Example:

* 7 (members) / 4 = 1.75

> * 1.75 * 2 = 5.25

Reviewer 2: times 3 you mean?

Reviewer 3: Fixed. Thanks for the catch, @lavalamp!

their departure.

The committee can choose to accept a member's resignation and carry out
offboarding procedures at any time following the written notice.

### Removal

<!--

STEERING CONSIDERATIONS

Removal

- [ ] Special election for specific removal scenarios
  - What is the perception of having a committee remove a representative that was elected by the community?
- [ ] Codify the voting process and channels for removal scenarios
- [ ] Consider CoCC guidance on removals as prior art: https://github.com/kubernetes/community/blob/master/committee-code-of-conduct/charter.md#removal
- [ ] Consider prior art in changes to maintainership e.g., SPIFFE/SPIRE (https://github.com/spiffe/spire/blob/main/MAINTAINERS.md#changes-in-maintainership)

COMMUNITY CONSIDERATIONS

No Confidence

- [ ] Provide guidelines on "no confidence"
  - Process transparency could be considered inversely proportional to the importance of these guidelines

Removal

- [ ] Scope of removal of members (one, many)
- [ ] Precedence for removal scenarios
- [ ] What details of removal scenarios should be public/private?
  - Call for removal?
  - Reason for removal?
  - Vote on removal?

>   - What are the public artifacts of the removal process?

Reviewer 9: If the `Removal` is made a public artifact, it would be good to somehow loop in the project moderators.
I would say the moderators do not need to know every detail but it would be good they be in the loop whether by the the steering committee or the coc committee

- [ ] What details of election results should be public/private?
- [ ] Consequences of removal scenarios
  - Are individuals ineligible for re-election?
    - Should community members have an opinion in this decision?
- [ ] When is removal appropriate?
- [ ] How should CoC concerns be considered in the removal process? Should they be?
- [ ] Guard against "popularity contests" or "burning at the person" in committee voting scenarios
- [ ] Are there antitrust implications in the removal process?
- [ ] Special elections may not have representative turnout
  - Define guidance on acts and thresholds to consider in a vote for removal
- [ ] Is removal an extraordinary power to grant?
  - "Policies like this have to work for the good of the community both when used well and when used poorly."
- [ ] Define removal process guidelines
  - "brief rationale for removal" + short time + "brief rebuttal"
  - Offer resignation as an option

-->

### No confidence

> In the event a Steering Committee member receives a vote of no confidence from

Reviewer 4: Possible clarifications:

Detail: Is the no-confidence vote public?

Detail: Is the rationale for the no-confidence vote public?

Detail: Does the subject of the no-confidence vote get to offer a rebuttal?

Detail: How is a no-confidence vote called?  One person, or one-person plus a seconding of the motion?  Something else?

Detail: Does the subject of the no-confidence vote get time between someone calling for a vote of no-confidence and the actual to think/consider/discuss?

two-thirds of the remaining Steering Committee members, that person must either

> resign or call a special election in which the project community will vote on

Reviewer 4: I appreciate that this swings pretty hard in the "more transparent" direction, but (and this is just my opinion, which I won't fight too hard for) I think it goes too far and drags things out in a way that isn't useful.

"must resign" is an oxymoron.  Presumably a straw-poll will be taken amongst steering.  If the person in question knows they will lose that vote, and still chooses not to resign, we don't have a removal process, we have a recall process.  This will require significantly more time, energy, and drama - likely including campaigning which will do more damage to all parties INCLUDING the project writ large.

If that is the path you all want to take, I will stop debating it.  My goal is not to make it all more difficult.

Reviewer: @justaugustus @parispittman i think we need to talk about this. There was a simpler path in the discussion above https://github.com/kubernetes/steering/pull/248#discussion_r911469884 that we should noodle on.

Reviewer 10: I'm inclined to agree with @thockin and @dims. Wrangling the normal SC election is already a time-intensive process. Doing it ad-hoc would REALLY draw things out. I think its worth discussing a more streamlined path.

Reviewer 3: Sounds good! This was an attempt to incorporate some of the feedback on what was initially written and thoughts about special elections (like https://github.com/kubernetes/steering/pull/248#issuecomment-1171036764) and then blend the two approaches.

I'll noodle on simpler paths (like https://github.com/kubernetes/steering/pull/248#discussion_r911469884) while more comments roll in.

Reviewer 11: +1 - I don't think the special elections route is useful and drags out a process that realistically needs to be concluded as soon as possible.

Reviewer 4: For the record, when I argue for "transparency", I am not calling for a trial or even evidence, really.  Simply a statement of rationale and maybe a rebuttal.

My mental model is something like:

* person X proposes a motion for a vote of no confidence, with a 100 words or less statement (public)
* person Y seconds
* the motion is carried
* the subject of the motion, if they so desire, has N days to make a 100 words or less rebuttal (public)
  - N is not less than 7 and not more than 28, as determined by the committee to accomodate travel or other plans person Z may have
* after N days a public vote of the committee is held

No special election, no demanding evidence.  It's not a trial.  I'm 50/50 on public comment, but if so, it is non-binding and voluntary (not sure if anonymous or not?).

If the community overall thinks there was injustice done and that the committee does not represent their values, there are regular elections.

Reviewer 2: The only thing I'd add to that is that the timeline should give the subject time to produce a rebuttal well before N, it would not go well to produce a rebuttal the day before the vote. But that's (hopefully obvious to the subject) just strategy.

> whether that person remains or is removed from the Steering Committee. A

Reviewer 2: I think you should define what happens in case of a vacancy separate from how a steering member gets removed, because there's multiple causes of that and they should likely be treated the same way.

For example, if there's less than 3 months (random number) until the next election, there probably needn't be a special election or even a back-fill.

Reviewer 2: Vacancies actually are already defined below, so this passage is confusing, why would the outgoing member call an election in this case only? probably I'm not reading it the way you intended.

> majority of eligible voters participating in the election must vote in favor of

Reviewer 2: This special election (I think "recall" is a better word) sounds very expensive, personally I'd just require unanimity from the board (that's # permanent seats - 1 yeses), notice to the community with some vague rationale, and time between notice and the meeting where the vote is held, at least two weeks. More discussion about this is in another comment thread.

Reviewer 2: Also due to our voting system, 50% of voters voting to retain a board member is quite likely a much higher fraction than what got them elected in the first place. If you keep this recall election, I think I would flip it around and make it a supermajority: 60% must vote in favor of ejection.

the individual remaining; otherwise, they will be removed. The individual's
participation in the Steering Committee will be suspended during the pendency
of the special election.

## Voting

<!--

COMMUNITY CONSIDERATIONS

Voting

- [ ] Better define voting requirements in specific scenarios
  - Consider 5/7, as 4/6 is a simple majority of the entire committee
  - Clarify requirements respective to size of committee and round up e.g., 7 * 2/3 = 4.6666..., which would be 5 votes out of 7
  - Are there scenarios where unanimity should be considered?
  - [Stephen] What about a combination of both?

-->

In the course of the committee's operations, members will be expected to vote
on decisions within the body's purview.

These votes may proceed on agreed-upon platforms by the committee, such as:

* a pull request
* an issue
* a Steering Committee [meeting](#meetings)
* a mailing list

Unless otherwise specified by a process, the requirement for passing a vote is

> a **_two-thirds supermajority of the

Reviewer 2: I think this is refactoring rather than defining for the first time, or I would complain that this is an excessively high bar and > 50% seems more natural for routine business.

Reviewer: @parispittman @justaugustus - need to address this?

Reviewer 2: I don't think you need to fix this in this PR.

Reviewer 6: Yeah, this is a change to current process, and usually a simple majority has been fine.

[fixed membership of the committee](#composition)_**.

Example:

* 7 (members) / 3 = 2.333...
* 2.333... * 2 = 4.666...
* Round up to the nearest whole number (5)

> * 5 members in attendance would be required to vote during a meeting

Reviewer: @parispittman @justaugustus - need to address this?

> * 5 members in attendance would be required to vote during a meeting

Reviewer 2: I don't think you need to fix this in this PR.

### Abstention

For any self-elected reason, members of the committee may decide to abstain
from a vote.

> Abstaining members will only be considered as contributing to quorum, in the

Reviewer 2: Abstaining is effectively a no, since you need 5 votes to pass anything on a 7 member committee -- as you've set it up the denominator never changes; usually abstaining changes the denominator in my understanding.

I believe quorum hasn't been defined above in this document yet.

Reviewer 2: oh, it's defined below, I would move that section up.

> event that a vote is called in a meeting.

Reviewer: we should call out when we don't count the abstaining members as well. (guessing they won't count towards `two-thirds supermajority`) in lines 181-183 above

Reviewer 3: Will attempt to adjust the language to better reflect!

* 4 members in attendance would be required to meet

Quorum **to vote in a meeting** is a **_two-thirds supermajority of the
[fixed membership of the committee](#composition)_**.

Example:

* 7 (members) / 3 = 2.333...
* 2.333... * 2 = 4.666...
* Round up to the nearest whole number (5)
* 5 members in attendance would be required to vote during a meeting

## Elections

Every year, the Steering Committee holds a general election for open seats with
[SIG Contributor Experience][sig-contribex] owning the process.

SIG Contributor Experience may delegate this responsibility to a subproject.

Our [election policy document][general-elections] covers the details for how
this works.

Policy and operations of special elections initiated by the Steering Committee
for other reasons (as yet to be determined) may be handled by some agreed-upon
[process][changes] by the committee.

[general-elections]: /elections.md
[sig-contribex]: https://git.k8s.io/community/sig-contributor-experience

### Vacancies

In the event of a resignation, removal, or other loss of an elected Steering 
Committee member, the candidate with the next most votes from the previous 

> election will be offered the seat. This process will continue until the seat is 

Reviewer 2: I would set a limit on how many people can be taken up this way, otherwise eventually you get to candidates who the community has basically completely rejected. Maybe something like no more than two people can be obtained this way, and anyone obtained this way must also be in the top 25% of that election.

Reviewer 2: oh, I see you moved this text. Well, maybe this isn't the best time to fix this (and my subsequent comment).

Reviewer 2: Well, actually, I do think this needs to change given the board can eject people. We do not want the board to go down the list ejecting people until they get the person they like.

Reviewer 12: Maybe only one seat can be filled from the list of candidates based on their votes, if they are then ejected an election must happen, that way it allows for quick/easy resolution of a vacancy, but heavily disincentivizes a Nixon/Watergate situation.

filled. 

In case this fails to fill the seat, a special election for that position will 

> be held as soon as possible. [Eligible voters](https://git.k8s.io/steering/elections.md#eligibility-for-voting) from the most recent election 

Reviewer 2: Add a time limit, just leave the seat vacant if there is only a month until the next regular election. Personally I'd go as high as six months given how expensive elections are. But it would be silly to have a special election for a period of time when the board won't even meet.

If the board is missing enough people that it can't make the voting quorum then there really does need to be an emergency special election.

Reviewer 9: I tend to agree with this. Perhaps an emeritus can fill the vacant spot temporarily in the example of a vacancy of a months or more (but not more than say 6 months)

Reviewer 2: I don't think I would do that as it could select someone the community deliberately voted out.

I think the board should work fine with 1 unfilled seat, it just makes passing things slightly harder. With 2 unfilled seats everything has to be unanimous, which is iffy. With 3 unfilled seats they can't do anything at all.

Reviewer 9: Very good points, thank you

- Consider explicitly mentioning SIG ContribEx's Elections subproject, once it
exists.

-->

Every year, the Steering Committee holds a general election for open seats with
[SIG Contributor Experience][sig-contribex] owning the process.

Our [election policy document][general-elections] covers the details for how
this works.

In the instance that a general election is called as a result of the
[dissolution](#dissolution) of the current committee, the following rules will
apply:

* members of the current committee will not be allowed to run in the new
  election
* a new committee will be bootstrapped with:
  * the top four candidates serving a two-year term
  * the next three candidates serving a one-year team

Policy and operations of special elections initiated by the Steering Committee
for other reasons (as yet to be determined) may be handled by some agreed-upon
[process][changes] by the committee.

[general-elections]: /elections.md
[sig-contribex]: https://git.k8s.io/community/sig-contributor-experience

### Vacancies

In the event of a resignation or other loss of an elected steering committee
member, the candidate with the next most votes from the previous election will

> be offered the seat.  This process will continue until the seat is filled.

Reviewer 8: It makes sense to fill the seat if the candidates below received a good number of votes (ie top 2-3 runners up with X% of the vote min) but doesn't quite makes sense if you just go down a list infinitely and appoint someone with barely any votes instead of just having a new election.

Reviewer 8: (Realizing that this is the existing policy but my comment stands esp given the new additions to the document)

Reviewer 2: 3 or 4 people have said similarly, but the old comment seems to have gotten detached from the current revision.

Reviewer 3: Linking @jberkus' [review](https://github.com/kubernetes/steering/pull/248#discussion_r914285986) to reconnect the history:

```suggestion
Committee member, the next most preferred candidate from the previous 
```

> Per below, I'd like to hold off on revising this section because I wanted to do it through the Elections Subproject.  However, if we are revising it, the phrase "next most votes" needs to go away, because it's not accurate.  We use preference elections.

@parispittman:

> agree with you. lets not change this language right now.

Reviewer 2: This section didn't matter before since there was basically no way to get a vacancy, but now there are several ways. I think you can fix this with a single additional line, "A maximum of one board member may be selected this way between elections." I would add more, but I think that's the minimum right now.

Reviewer 2: Oh, I see what happened -- @justaugustus that's not the right comment thread, let me find it for you.

Reviewer 2: Thanks!

Reviewer: Marking this as resolved.

redetermined at the time of the special election. A committee member elected in
a special election will serve out the remainder of the term for the person they
are replacing, regardless of the length of that remainder.

[voter-eligibility]: /elections.md#eligibility-for-voting

### Resignation

If a committee member chooses not to continue in their role, for whatever
self-elected reason, they must notify the committee in writing.

### Dissolution

If committee members believe that the committee is no longer able to act in
accordance with the above [mission](#mission), the committee may dissolve by a
few methods:

* Dissolution by vote
* Dissolution by loss of committee

See [elections](#elections) for special criteria in the instance of a general
election triggered by dissolution.

#### Continuity

If the committee is dissolved completely by any mechanism, selected members
(selection described below) agree to be "continuity advisors" and attend
meetings of the subsequently elected incoming committee in a non-voting
capacity for 6 meetings or until the incoming committee dismisses them,
whichever is first.

> ##### Selection

Reviewer 8: I think it could be difficult to mandate that members of a dissolved committee commit to serve in this capacity. Is this para more about the new committee _trying_ to get continuity as opposed to trying to mandate participation from outgoing members?

Reviewer 2: Yes, maybe we should add, at the end, "if a selected former member isn't willing or able to perform this duty then the incoming board may choose a replacement. If there are no willing members then the new board will just have to all watch the last N meeting recordings."

At least two (2) and not more than three (3) members shall be selected for
this duty.

* One shall be the longest-serving outgoing committee member. If multiple, the
  incoming committee may select between them.
* The second shall be selected by the incoming committee at their discretion.
* The incoming committee, at their discretion, may select a third.

#### Dissolution by vote

Dissolution requires an affirmative vote of a **_two-thirds supermajority of
the [fixed membership of the committee](#composition)_**, and must be unanimous

> when the committee has four (4) or fewer members.

Reviewer 8: I'm a little confused about how this sentence squares with the one below (loss of committee). It doesn't need to be unanimous in all cases under 4 members bc if the committee (starting with 7 members) ends up with 4 vacant seats and 3 members it automatically dissolves.

Shouldn't it just say:
"and must be unanimous when the committee has four (4) members. See dissolution by loss of committee for dissolution when less than 4 members"

or something.

Reviewer 2: Yeah, I don't understand why people would do this vote instead of mass-resigning.

Reviewer 3: We'll work on dissolution process in https://github.com/kubernetes/steering/issues/250.

Example:

* 7 (members) / 3 = 2.333...
* 2.333... * 2 = 4.666...
* Round up to the nearest whole number (5)
* An affirmative vote of 5 members would be required to dissolve the committee

The committee must specify a date of dissolution.
If the committee is dissolved, all seats are vacated on the date specified.

#### Dissolution by loss of committee

If at any time a **_majority of the
[fixed membership of the committee](#composition)_**'s seats are vacant, the entire committee will be dissolved.

Example:

* 7 (seats) / 2 = 3.5
* Round up to the nearest whole number (4)
* If 4 committee seats are vacant, the entire committee will dissolve

The number of vacant seats will be evaluated prior to any replacement of
previously-seated members.

### Removal

#### No confidence

A Steering Committee member may be removed by a three-fourths vote of no
confidence from other Steering Committee members.

The call for a vote of no confidence will happen in a public Steering Committee meeting and a

> GitHub issue in kubernetes/steering by a current member and must be seconded by

Reviewer 13: Respectfully, there are cases where the specifics of a CoC violation cannot be revealed to protect a victim and/or reporter. The action taken can be public, but the actual specifics around the violation should not be revealed in public without careful consideration of the safety of folks involved.

> GitHub issue in kubernetes/steering by a current member and must be seconded by

Reviewer 14: > Respectfully, there are cases where the specifics of a CoC violation cannot be revealed to protect a victim and/or reporter. The action taken can be public, but the actual specifics around the violation should not be revealed in public without careful consideration of the safety of folks involved.

Then it should be possible to point to the public censure against the steering member.  A CoC violation that does not result in a public censure of some kind seems like a thin pretext for removing the member from office.

> GitHub issue in kubernetes/steering by a current member and must be seconded by

Reviewer 13: Ah, I misunderstood your point, then. I thought you were saying the specifics of the violation were needed for the community to judge, rather than saying there needed to be an acknowledgement that there was a violation and censure.

> GitHub issue in kubernetes/steering by a current member and must be seconded by

Reviewer: @deads2k A censure is likely to work when the person censured is likely to change what they do, for those who are likely to continue the same response to a situation, how many censures are enough? Why put community through the harm again and again?

> GitHub issue in kubernetes/steering by a current member and must be seconded by

Reviewer 3: I do think it would be worthwhile for the other (non-affected) members of Steering to produce a _short_, general statement providing some context for the vote of no confidence.

Personally, as a community member and voter, I'd want some level of explanation:

- to inform my future votes, and especially:
- in the absence of any restrictions from that member running in future elections (current state)

I do not however, think it would be useful to mandate:

- a format in which community feedback should be received by the committee
- that that feedback be committed to public fora

I feel that we as Steering members find ways to catalyze private feedback.

If that is a misaligned assumption on my part, then it's something that should discussed as part of https://github.com/kubernetes/steering/issues/250 (which will iterate on feedback on elections process and dissolution of committee).

> GitHub issue in kubernetes/steering by a current member and must be seconded by

Reviewer 14: > I do not however, think it would be useful to mandate:
> 
> * a format in which community feedback should be received by the committee
> * that that feedback be committed to public fora

How will the community be able to inspect and understand why their elected representative was removed if we do not require that the reasons for the removal be public?

> GitHub issue in kubernetes/steering by a current member and must be seconded by

Reviewer 14: > Personally, as a community member and voter, I'd want some level of explanation:

I think it's also important to see how each steering member voted.  I think it's possible that some community members would disagree with removing a steering member and would want to explicitly down-vote every standing steering member who voted for removal.

> GitHub issue in kubernetes/steering by a current member and must be seconded by

Reviewer 3: Added providing some short statement to set context on the vote in https://github.com/kubernetes/steering/pull/248/commits/7564e40a553263a936b337201f687e45a0e7d356.

another current member.

Once a vote of no confidence has been called, the committee will notify the
community through the following channels:

* the [community mailing list][dev-list]
* the [Steering Committee public mailing list][steering-public-list]

This notification should include a link to the aforementioned GitHub issue.

> There will be a period of one week for members of the community to reach

Reviewer 14: ```suggestion
There will be a period of four weeks for members of the community to reach
```

One week looks short. The rule-making process is four weeks without unanimous agreement to vote early, the meeting cadence is monthly, this stands out as an unusually short time frame. Four weeks lines up well with both policies.

Reviewer 3: Updated to two weeks to reflect the conversation about comment periods from a previous review:

From @lavalamp in https://github.com/kubernetes/steering/pull/248#discussion_r911469884:

> I was going to say a week seems OK, but it'd really suck to have a board pull this on a member who is on vacation or something and can't respond. Personally I'd probably go longer, maybe 2 weeks.

From @dims in https://github.com/kubernetes/steering/pull/248#discussion_r911473383:

> @lavalamp feels reasonable to me. I'd stick to private feedback mechanism, just email steering private so everyone on steering sees it including the person in question.
> 
> yes you are right on holidays/vacations ... so 2 weeks.

out to Steering Committee members to provide feedback, which can include
non-binding votes.

Community members may provide feedback by the following methods:

* commenting on the GitHub issue
* replying to the notification email
* sending an email to the
  [Steering Committee private mailing list][steering-private-list]
* sending a message to individual committee members

After this feedback period, Steering Committee members must vote on the issue
within 48 hours.

[dev-list]: mailto:dev@kubernetes.io
[steering-private-list]: mailto:steering-private@kubernetes.io
[steering-public-list]: mailto:steering@kubernetes.io

## Voting

In the course of the committee's operations, members will be expected to vote
on decisions within the body's purview.

> These votes may proceed on agreed-upon platforms by the committee, such as:

Reviewer 8: Is this saying the vote will/can take place in multiple venues? "platforms"

Feels like it should take place in 1 for clarity.

Reviewer 3: @kikisdeliveryservice -- As an example, a vote may be initiated on a meeting, with the actual vote captured as an artifact of a PR or issue.

Would it be useful to say something around the vote being called for on any of the accepted platforms, but needing to be captured within an issue or PR (in the case of public votes)?

Reviewer 8: That clarifies it for me 🙂

> These votes may proceed on agreed-upon platforms by the committee, such as:

Reviewer 5: ... except that the SC does do some non-public business as well, particularly referring legal matters to the LF, and such business may require a vote, which would necessarily be non-public.

> These votes may proceed on agreed-upon platforms by the committee, such as:

Reviewer 3: > SC does do some non-public business as well

Agreed. Marking as resolved.

* Control and delegate access to and establish processes regarding
  project resources/assets<sup>[6](#footnote6)</sup>  

## Changes

In instances where a process is not already specified within this document,
changes to the Steering Committee charter will be considered according to the
processes set forth in the committee's [operations documentation][changes].

[changes]: /operations/changes.md

## Membership

### Composition

The Steering Committee is composed of seven (7) members.

### Elections

<!--

- Consider explicitly mentioning SIG ContribEx's Elections subproject, once it
exists.

-->

Every year, the Steering Committee holds a general election for open seats with
[SIG Contributor Experience][sig-contribex] owning the process.

Our [election policy document][general-elections] covers the details for how
this works.

In the instance that a general election is called as a result of the
[dissolution](#dissolution) of the current committee, the following rules will
apply:

* members of the current committee will not be allowed to run in the new

>   election

Reviewer 13: Perhaps noting that there *is* a plan for continuity below would be helpful; personally, I'd like to know that there's still a plan to provide support to new members (considering that existing steering members are unable to run).

I think the thing that isn't clear for me here, though, is whether those members are unable to run once or forever. Is the intention to say the steering members who were on the disbanded committee can never run again, or only for this election and then they're eligible for the next election?

Reviewer 2: I believe the intention is once, and I agree it'd be good to state that explicitly.

### Vacancies

In the event of a resignation or other loss of an elected steering committee
member, the candidate with the next most votes from the previous election will
be offered the seat.  This process will continue until the seat is filled.

In case this fails to fill the seat, a special election for that position will
be held as soon as possible. [Eligible voters][voter-eligibility] from the most
recent election will vote in the special election i.e., eligibility will not be
redetermined at the time of the special election. A committee member elected in
a special election will serve out the remainder of the term for the person they
are replacing, regardless of the length of that remainder.

[voter-eligibility]: /elections.md#eligibility-for-voting

### Resignation

If a committee member chooses not to continue in their role, for whatever
self-elected reason, they must notify the committee in writing.

### Dissolution

If committee members believe that the committee is no longer able to act in
accordance with the above [mission](#mission), the committee may dissolve by a
few methods:

* Dissolution by vote

> * Dissolution by loss of committee

Reviewer: +1

#### Dissolution by vote

Dissolution requires an affirmative vote of a **_two-thirds supermajority of
the [fixed membership of the committee](#composition)_**, and must be unanimous
when the committee has four (4) or fewer members.

Example:

* 7 (members) / 3 = 2.333...
* 2.333... * 2 = 4.666...
* Round up to the nearest whole number (5)
* An affirmative vote of 5 members would be required to dissolve the committee

The committee must specify a date of dissolution.
If the committee is dissolved, all seats are vacated on the date specified.

#### Dissolution by loss of committee

If at any time a **_majority of the
[fixed membership of the committee](#composition)_**'s seats are vacant, the entire committee will be dissolved.

Example:

* 7 (seats) / 2 = 3.5
* Round up to the nearest whole number (4)
* If 4 committee seats are vacant, the entire committee will dissolve

The number of vacant seats will be evaluated prior to any replacement of
previously-seated members.

### Removal

A committee member may be removed from the committee by the other committee
members. The member should be given the opportunity to resign before they are
removed.

> Removal should be considered for the following reasons:

Reviewer 7: Should the "should" be a "must"?

With the current language the steering committee could chose another reason (like they don't like the persons hair do... to provide a fuzz-like input).

Given the "should" along with no requirement to share the reason there could be something the broader community would not approve of that is kept private. In fact, in removal there is nothing that says the reason even needs to be communicated to the removed party.

How much lee-way should the steering committee have over deciding removal of someone who was voted in? What transparency requirements should be here over the reason for that? I ask about transparency so the broad community can be well informed.

Reviewer: thanks for your input Matt.

Reviewer: @mattfarina you must be kidding me at this point. So i am not going to engage.(like they don't like the persons hair do... to provide a fuzz-like input).

If you use an iota of your skills to protect folks in the community, that would be a better use of your time.

Reviewer: > If you use an iota of your skills to protect folks in the community, that would be a better use of your time.

That was probably a bit too harsh. Apologies Matt.

> Removal should be considered for the following reasons:

Reviewer 2: I think this list is problematic and should be omitted, and a better list can be written in the future. For now I think the only reasonable requirement is that the reason be public so the community can judge the result. I think everyone is too anchored at the moment to produce a bias-free list.

Specifically, to be helpful the list here should state that: the reason given should be (or need not be) similar in severity; and also give examples of insufficient reasons. You need both positive and negative test cases to be sure your code is doing the right thing!

Reviewer 13: While I agree that the list could use to be reviewed and built up a bit (or noted as not comprehensive), I'm also aware that there are reasons that may need to be private to avoid violations of privacy laws, for example.

Reviewer 4: agree.  I don't think this list particularly helps

Reviewer: i agree, +1 to remove the list.

Reviewer 2: > reasons that may need to be private to avoid violations of privacy laws

In that case, the board needs to say "we have to eject member X and we can't explain the reason because it would put us in legal jeopardy".

I personally think that board members should carry liability insurance (honestly probably the CNCF should be paying for it, but if not, self-funded). And anything serious enough to kick someone off should be serious enough to defend in court.

This example has been mentioned several times in the comments here and I hate to say it, but I won't believe it unless I hear it from a lawyer. I think being a board member of an organization this size should come with reduced rights to expectation of privacy, and that crimes meriting eviction should also merit publicity. Otherwise it is too easy to abuse this system, you can do it with like only a 3 person conspiracy. 3 person conspiracies are actually possible in real life. (I'm not a lawyer and this is not a legal opinion, just a statement of what I'd want a lawyer to talk me out of to accept this reason.)

Reviewer: 😢 unfortunately!

Author: steering members agree with you re: insurance https://github.com/kubernetes/funding/issues/29 but it's not 100% complete yet

Reviewer 13: > > reasons that may need to be private to avoid violations of privacy laws
> 
> In that case, the board needs to say "we have to eject member X and we can't explain the reason because it would put us in legal jeopardy".
> 
> I personally think that board members should carry liability insurance (honestly probably the CNCF should be paying for it, but if not, self-funded). And anything serious enough to kick someone off should be serious enough to defend in court.
> 
> This example has been mentioned several times in the comments here and I hate to say it, but I won't believe it unless I hear it from a lawyer. I think being a board member of an organization this size should come with reduced rights to expectation of privacy, and that crimes meriting eviction should also merit publicity. Otherwise it is too easy to abuse this system, you can do it with like only a 3 person conspiracy. 3 person conspiracies are actually possible in real life. (I'm not a lawyer and this is not a legal opinion, just a statement of what I'd want a lawyer to talk me out of to accept this reason.)

I can understand that. The concern as I understand it is this is a volunteer community board versus a paid organization. While I prefer publication of reasons of eviction both for the community's health as well as for an assurance that no one is taking advantage of the community, my (fairly) limited experience with volunteer boards is there is a concern around privacy laws, and often the generic wording of separation is the choice. Is it ideal? No, not to me. But it's definitely something that we have to think about (and I'm also not a lawyer! So take my thoughts with a grain of salt considering I haven't had enough tea yet today...). In the end, the steering committee answers to and serves the community, and I agree that, where possible, providing reasons are healthy for the community itself.

Reviewer 2: @nimbinatus I heard "asymmetric lawyering capabilities", therefore "avoid nuisance lawsuits", and I can definitely appreciate that. But I highly doubt omitting a reason will achieve that goal, and it might even make such a lawsuit more likely! It might make it worse if it creates the appearance that the board doesn't believe it is doing something legally defensible!

Reviewer 13: > @nimbinatus I heard "asymmetric lawyering capabilities", therefore "avoid nuisance lawsuits", and I can definitely appreciate that. But I highly doubt omitting a reason will achieve that goal, and it might even make such a lawsuit more likely! It might make it worse if it creates the appearance that the board doesn't believe it is doing something legally defensible!

Very fair point. All the more reason to ensure a lawyer reviews the comms if we have that availability (would the CNCF pay for this? Do we have someone on retainer? I would expect that to be true.), and all the more reason to have a clear documentation guide for explicit removals versus people voluntarily stepping down. I'm assuming that documentation guide is elsewhere. In other volunteer organizations I've been part of, there have been cases where revealing a reason was determined to likely lead to revealing identities and/or retaliation against a victim, reporter, or a voting board member. Therefore the reason the vote was called and the vote itself (and the discussion) was documented and sealed as proof against a lawsuit while maintaining privacy and safety of others, and community members were given a generic note. The concern comes from my own experience on those kinds of boards and organizations; again, that experience is limited, not comprehensive, so it's my two cents. I trust that the steering committee is considering both of our feedback when reviewing these docs.

Reviewer 2: I agree it's a terrible dilemma to choose between protecting a 3rd party and demonstrating sufficiently skeptical truth acquisition ability. Fortunately I think these rules don't have to take a position on this right now.

* The member has been found to have committed a code of conduct violation.

> * The member is convicted of a felony.

Reviewer 2: I think this is problematic due to not specifying the country or legal system, they are not all equally just. Did a lawyer look at this language? Felony/misdemeanor seems not to be a universal distinction in legal systems from a very quick search.

Reviewer 15: I think this list mostly serves to be examples of cases, rather than "the only cases" where this might happen. 

That being said, I would generally agree with this being a bad example (or at least in need of nuance). Laws differ around the world and don't necessarily align with our values (e.g countries where queerness is persecuted or if someone was persecuted for protest or searching for healthcare).

Reviewer 2: Yes, that was my main point, thank you for clarifying.

(I have another comment stating why I think some negative examples are also necessary)

Reviewer 7: This is a great point. It seems to be US centric (or west centric) where k8s is global.

Reviewer: discussion below agree, https://github.com/kubernetes/steering/pull/248#discussion_r915034333 we want to just delete this list

Author: +1 to remove

Reviewer 10: +1 to remove as well

Reviewer 3: Removed.

* The member has been completely out of contact for more than 30 consecutive
  calendar days without having made prior arrangements.

> * The member has explicitly, publicly violated the privacy of a community

Reviewer 2: This is weirdly specific, how would this not be a CoC violation? I would remove this.

Reviewer: @lavalamp guessing this is outside of community spaces?

Reviewer: (then CoC can't step in)

Reviewer 2: @dims maybe? It's not what it says, if that's what it means it should be clear? (and then why not other aspects of the CoC also?)

If we want an extra-special CoC for board members, I don't think that's something that can be worked out in a short amount of time while people have an example in mind.

Reviewer: agree, https://github.com/kubernetes/steering/pull/248#discussion_r915034333 we want to just delete this list

* Quorum to meet is 1/2 members attending (eg: 6/12, 6/11, 4/7)
* Quorum to vote in a meeting is 2/3 members attending (eg: 8/12, 8/11, 5/7)

## Elections

Every year, the Steering Committee holds an election for open seats with SIG 

> Contributor Experience owning the process. Our [election policy document](link) 

Reviewer 16: We need to add the link here.

Author: 🤦🏻‍♀️ thanks for that catch @jdumars!

covers the details for how this works.

> ### Vacancies and Removal

Reviewer 17: More an optical nit
```suggestion

### Vacancies and Removal
```

In the event of a resignation, removal, or other loss of an elected Steering 
Committee member, the candidate with the next most votes from the previous 
election will be offered the seat. This process will continue until the seat is 
filled. 

In case this fails to fill the seat, a special election for that position will 
be held as soon as possible. [Eligible voters](https://git.k8s.io/steering/elections.md#eligibility-for-voting) from the most recent election 
will vote in the special election (ie: eligibility will not be redetermined at 
the time of the special election). A committee member elected in a special 
election will serve out the remainder of the term for the person they are 
replacing, regardless of the length of that remainder.

> #### Removal

Reviewer 17: ```suggestion

#### Removal
```

> A Steering Committee member may be removed by a two-thirds vote of the other 

Reviewer 14: Does removal automatically make an individual ineligible for re-election?  I think we should allow them to re-run.  Allowing someone who has been removed, to run for office again allows the community at large to decide whether or not that person should hold office.

Reviewer 6: My reading of the current language would be no, it would not automatically make you ineligible to run in the next full election.

Reviewer: same

> A Steering Committee member may be removed by a two-thirds vote of the other 

Reviewer 14: Are there any limitations or guidance for when the steering committee should hold a removal vote?  Any defined threshold for committee members to consider when deciding to remove a person from office?

"high crimes and misdemeanors" is vague and defined separately, but also makes it clear that the judgement about impeachment (equivalent to removal?) is about whether a criminal act has been committed versus a distasteful act or simply being unpopular among peers.

Reviewer 18: > Are there any limitations or guidance for when the steering committee should hold a removal vote? Any defined threshold for committee members to consider when deciding to remove a person from office?
> 
> "high crimes and misdemeanors" is vague and defined separately, but also makes it clear that the judgement about impeachment (equivalent to removal?) is about whether a criminal act has been committed versus a distasteful act or simply being unpopular among peers.

While I don't disagree that there should be some type of guidance or limitations, I think comparing it to impeachment is a bit excessive here...  Setting the bar at criminal acts is way to low for a body that is supposed to `Define, evolve, and defend the non-technical vision / mission and the values of the project`. I think we *must* expect the Steering Committee to hold itself accountable to the community by adhering to those defined values (https://github.com/kubernetes/community/blob/master/values.md). We also shouldn't expect that a sitting member of the Steering committee be immune from things like Code of Conduct violations and enforcement, would we?

Reviewer 19: > While I don't disagree that there should be some type of guidance or limitations, I think comparing it to impeachment is a bit excessive here... Setting the bar at criminal acts is way to low for a body that is supposed to Define, evolve, and defend the non-technical vision / mission and the values of the project. I think we must expect the Steering Committee to hold itself accountable to the community by adhering to those defined values (https://github.com/kubernetes/community/blob/master/values.md).

Agreed

> We also shouldn't expect that a sitting member of the Steering committee be immune from things like Code of Conduct violations and enforcement, would we?

Not at all. The list at https://github.com/kubernetes/community/blob/master/committee-code-of-conduct/charter.md#removal could be useful prior art in this area to consider.

Reviewer 6: As I mentioned https://github.com/kubernetes/steering/pull/248#discussion_r911317290, the intent is not to address code of conduct issues, but if, for any reason, the committee has no confidence in a member and feels that member is unfit to serve.

Reviewer 11: Members of the Steering Committee have previously [raised issue with the idea of being beholden to the CoCC in the same way other members of the project are.](https://github.com/kubernetes/community/pull/6247) 

I do not agree with the objections raised in the linked PR, but I do respect Steering's right to self determine their processes.

As a result I do not think Steering needs to consider any prior art in the Code of Conduct Committee's processes. We are independent bodies and are both free to set our own process independent of one another.

I see no problem with the wording or process presented here by SC. 👍, +1, /lgtm.

Reviewer 14: > when a member has no confidence in the leadership of the individual in question which could encompass most of your examples. do you think that would make it clear?

To a skeptical person, "no confidence" reads as, "for any reason someone wants" versus a guideline that indicates boundaries between taste, preference, opinion, and behavior.  I think a guideline that focuses on behavior and actions will be easiest to explain to observers of the process and easiest for the committee to consider when casting their votes.

Reviewer: @deads2k thanks for the feedback! we'll see what we can do.

Reviewer 2: I think the more transparent the process, the less important this guideline is (and vice versa).

Reviewer 20: +1 @lavalamp

> A Steering Committee member may be removed by a two-thirds vote of the other 

Reviewer 14: Quickly running the math on this, the outcome with seven members appears subpar: 4/6 is the same as simple majority vote.  The voting thresholds were developed with the size of the committee in mind and modeling this as a full committee vote where the removal target also gets to vote produces a better outcome where 5/7 members of the committee must agree, which matches the threshold for existing votes.

Reviewer 10: General practice has been to round up - this would be 5/7, from the charter on quorum:

```
Quorum
Steering committee members are generally expected to attend every meeting. We use the following guidelines to determine whether we have reached quorum and are able to proceed with a meeting.

Round quorum up if total number of members is odd
Quorum to meet is 1/2 members attending (eg: 6/12, 6/11, 4/7)
Quorum to vote in a meeting is 2/3 members attending (eg: 8/12, 8/11, 5/7)
```

Reviewer 14: > General practice has been to round up - this would be 5/7, from the charter on quorum:

The exact phrasing in this PR currently says 

> a two-thirds vote of the other Steering Committee members

So if there are 7 members, and 1 is being targeted for removal, the base number would become 6.  If the PR was rephrased to be 

> a two-thirds vote of the Steering Committee members

then it would be 5/7.  Can we get it rephrased to make this clear?

Reviewer 18: I don't think it should be codified specifically as an x/y, since then if for any reason the steering committee size would change over time, this number would need to be modified...  Maybe clarify a bit more that the 2/3 implies rounding up the number of votes needed when the number of members is not a multiple of 3?

Reviewer 14: > I don't think it should be codified specifically as an x/y, since then if for any reason the steering committee size would change over time,

I agree.  I think if the current intent is to be 5/7 instead of 4/7, that can be achieved by phrasing the requirement as 

> a two-thirds vote of the Steering Committee members

As currently stated, the threshold is 4/7 because the seventh was excluded from the vote.

Author: agree that we can clear this up, thinking, will come back with a change and ping y'all here.

> A Steering Committee member may be removed by a two-thirds vote of the other 

Reviewer 18: > The steering committee has both private and public meetings. Removal should be required to be a public meeting, with public reasoning, and a public vote.

Considering that the removal process could become quite contentious and we can't foresee all of the possible situations that might lead to the removal process being invoked, especially in the case of things like CoC violations, I don't think we can expect that it will always be possible to conduct the removal process in a public forum.

That said, I do believe that there should be a public artifact produced that is made available providing relevant details to the community regarding the removal process.

> A Steering Committee member may be removed by a two-thirds vote of the other 

Reviewer 15: There are sometimes legal reasons to not disclose _why_ something has happened. CoCC actions are notably not public for this reason too.

> A Steering Committee member may be removed by a two-thirds vote of the other 

Reviewer 2: Generally speaking things like this have some sort of defined initiation procedure, not just a resolution procedure. I would expect this to require some sort of public charge and some sort of minimum amount of time for a defense. I would expect e.g. at least two board members to agree on the charge before this resolution procedure is triggered.

Reviewer 2: That's a very powerful ability to give to the committee, and I think unanimity (sans person in question) would be a better bar than 2/3 rounding up.

I also think giving the person in question zero warning and zero time to even prepare a statement is not fair at all.

I would much prefer a charge + time + vote than this. If it's worth removing someone, it's worth a public spectacle, you get a lot more "this is who we are" milage out of that. And IMO if we're not willing to pay the cost of public drama then we don't actually want the person to be removed that badly. Charges can be vague if necessary.

Reviewer 2: Policies like this have to work for the good of the community both when used well and when used poorly. If we're ejecting someone for good reasons the community should appreciate the board's actions. If we're ejecting someone for bad reasons the community should be angry and vote out the board at the next election. The community needs something to evaluate in order to be a good check on the board's behavior.

Reviewer 20: Do we have prior art from other communities, and how it has worked for them?

This does seem like an extraordinary power to give the committee. It is the power to effectively undo the results of the election for a specific member. I can see reasons to remove someone - for example, code of conduct violation or dereliction of duty (e.g., not showing up to any meetings or participating). But without any criteria or even guidance in the appropriate use of this power, we risk silencing voices simply because they disagree with or are unpopular among the other steering committee members. This will lead to group think and is not a healthy state for our leadership and governance organizations.

Another option to achieve the same goal could be to allow the steering committee to initiate a recall election. This would put the decisions in the hands of the community, rather than in the small number of people on the steering committee.

In either case (removal or initiating a recall), given the small size of our steering committee, I believe a more appropriate committee voting threshold would be unanimity, excluding of course the subject of the vote and any abstentions.

Reviewer 4: I tend to agree with Daniel here, though I do not think we want a long, drawn out "hearing" with cross-examination etc.  Transparency is crititcal to trust.

I think "rationale for removal" + short time + "rebuttal" is fair.

If the target wants to avoid the attention and thinks they are likely to lose the vote, resignation is always an option.

Reviewer: @thockin The crux say if a person thinks what they are doing is totally right and completely oblivious of their effect on the community. They will stand their ground rendering the steering ineffective. What you and others are suggesting prolongs the agony and the harm.

NOTE that this is NOT a process to get someone out of the community. Just from their leadership position in steering. Nothing else. When the steering members do not have confidence in this person repeating a behavior and further expand the harm they are committing (And refusing to resign) there needs to be a mechanism to quickly deal with it.

There is MUCH MORE stuff we have to deal with as a community. This was not needed at this point in time. Unfortunately here we are trying to draft something because we had total confidence in abilities of those who we vote in to leadership positions and we have been throughly let down.

This is NOT something that happens every release cycle or year. this is a very drastic step and is not to be done lightly.

Reviewer 2: "very drastic and [won't] be done lightly" is in some tension with the (hopefully I misread, but) apparent desire to do this without letting the community at large know what is going on.

Reviewer 4: > What you and others are suggesting prolongs the agony and the harm.

I don't buy this.  Such an action DESERVES consideration and "cooler heads".  It needs to be (relatively) dispassionate.  I'm not saying this is a 6 month or even 6 week process.  I'm talking about a week or two.

> we have been throughly let down.

This statement is what I want to be careful of.  It is not dispassionate.  I don't think the  _process_ should be crafted in response to an individual, urgent situation, especially while heads are hot.  And look, you KNOW I am not saying "do nothing".  I'm just arguing that the caution of the process should be proportional to the gravity of the result.

> this is a very drastic step and is not to be done lightly.

Exactly.  I'd rather err on the side of caution and transparency and "fairness".  Just try to turn the situation around and imagine yourself in the hotseat.

Reviewer: @thockin @lavalamp ACK. will try to see what we could be doing in this process better. thanks for your input.

Reviewer: @thockin @lavalamp feels like steering would write something up, give it a time period, allow conversations to be had in the time in between and then VOTE.

The conversations would help the steering leaders to convince them on which way they should vote. 

How does that sound? May i suggest one week time frame?

Reviewer: I'll leave it to the PR author Paris to see if this is something that can be folded into this document.

Reviewer 2: The community should be informed at the beginning of the time period, so at the very least they can email opinions to steering members. I think public comments would be even better.

I was going to say a week seems OK, but it'd really suck to have a board pull this on a member who is on vacation or something and can't respond. Personally I'd probably go longer, maybe 2 weeks.

Reviewer: @lavalamp feels reasonable to me. I'd stick to private feedback mechanism, just email steering private so everyone on steering sees it including the person in question.

yes you are right on holidays/vacations ... so 2 weeks.

Reviewer 2: I can live with that. But let me point out that public comments create [common knowledge](https://en.wikipedia.org/wiki/Common_knowledge_(logic)), and private ones don't. That is, it's important for the community to know what the community thinks, not just for the steering committee. 

I do see the potential for abuse with public comments, but also we do have moderation (which seems very effective, thank you moderators!).

Reviewer: thanks @lavalamp

> A Steering Committee member may be removed by a two-thirds vote of the other 

Reviewer 6: > Removal should be required to be a public meeting, with public reasoning, and a public vote.

Votes are always public, but sometimes might be done async on an issue. As for reasoning, each member might have slightly different reasoning as is the case any time we vote on an issue.

I'll note that this section doesn't mention and isn't related to the code of conduct. CoC violations are handled by the Code of Conduct committee, not Steering. This would be if, for any reason, the committee has no confidence in a member and feels that member is unfit to serve.

> A Steering Committee member may be removed by a two-thirds vote of the other 

Reviewer 11: +1 to @endocrimes' reasoning. The Committees are chartered with a certain level of privacy in mind. 

Any public requirement for removal for a member would end up either as a popularity contest in the project or a burning at the stake of the person in question.

> A Steering Committee member may be removed by a two-thirds vote of the other 

Reviewer 14: A secret tribunal to remove someone who was elected as a community representative for secret reasons, with a secret vote is not in keeping with community values around transparency.  If the action is important enough to remove the person from office (code of conduct or otherwise), it's important enough to explain why that must be done, how it is in the best interest of the community, and how community representatives cast their votes.

> A Steering Committee member may be removed by a two-thirds vote of the other 

Reviewer 20: I agree secrecy is not in line with our values. In fact, there may be legal antitrust implications - we should ask LF legal about that.

> A Steering Committee member may be removed by a two-thirds vote of the other 

Reviewer: @deads2k @johnbelamaric please scroll down to see the discussion below with Daniel and thockin.

> Steering Committee members, with the impacted individual ineligible to 

Reviewer 2: It's kind of mathematically pointless to make them ineligible to vote, the meaningful question is whether they count in the denominator as above; it's easy to predict what their vote would do to the numerator.

### Vacancies

In the event of a resignation or other loss of an elected committee member, the
next most preferred candidate from the previous election will be offered the
seat.

A maximum of one (1) committee member may be selected this way between
elections.

In case this fails to fill the seat, a special election for that position will
be held as soon as possible.

[Eligible voters][voter-eligibility] from the most recent election will vote in
the special election i.e., eligibility will not be redetermined at the time of
the special election.

A committee member elected in a special election will serve out the remainder
of the term for the person they are replacing, regardless of the length of that
remainder.

[maximal-representation]: /elections.md#maximal-representation
[voter-eligibility]: /elections.md#eligibility-for-voting

### Resignation

If a committee member chooses not to continue in their role, for whatever
self-elected reason, they must notify the committee in writing.

### Removal

#### No confidence

A Steering Committee member may be removed by an affirmative vote of a
**_three-quarters supermajority of the
[fixed membership of the committee](#composition)_**.

> Example:

Reviewer 6: I'd like to suggest we remove all examples from this document. They are helpful and illustrative when debating, but now that we have specific language, I'd suggest we remove the examples.

Reviewer 2: I think the examples are vital for making it 100% clear the intent of the words.

Reviewer 19: I agree they should be kept to keep the meaning unambiguous

Reviewer 6: Is there language we could add or modify as far as the in-force language that would make the example unnecessary?

I feel like if we need an illustrative example in addition to the in-force language, that we might not have done a good enough job with the in-force language.. we shouldn't need to state the same thing twice for it to be clear.

Reviewer 2: Stating the same thing two (2) times in different ways is very common in documents like these in my experience. It demonstrates that the author didn't make a mistake.

Reviewer 6: Do you have examples of where this is done in a charter document like this?

Reviewer 2: Not off the top of my head, sorry. I don't understand why you think it is bad to include them? This isn't go code, redundancy doesn't seem bad, especially since there's no compiler :)

Reviewer 6: My concern would be, if you state the similar things twice but in different words, it leaves room for dispute on which words are actually the in-force language.

If there was a way to convey the information fully as one cohesive statement that didn't require saying it a second time, there's less opportunity for dispute in the future.

Reviewer 2: Yes, it would be very bad if the example didn't match the text. But the examples do match the text, so we won't have that problem?

Reviewer 6: If we have clear in-force text, then an example shouldn't be required.

Reviewer 3: I'm going to opt to resolve this convo on examples.
It sounds like the examples help remove ambiguity for readers (which was my hope).

If we want to come back at a future date and further strengthen the in-force text to remove examples, it should roughly be a no-op (intent-wise).

  of no confidence

The call for a vote of no confidence will happen in a public Steering Committee
meeting and must be documented as a GitHub issue in the committee's
[repository][steering-repo].

The call for a vote of no confidence must be made by a current member of the
committee and must be seconded by another current member.

The committee member who calls for the vote will prepare a statement which
provides context on the reason for the vote. This statement must be seconded by
the committee member who seconded the vote.

Once a vote of no confidence has been called, the committee will notify the
community through the following channels:

* the [community mailing list][dev-list]
* the [Steering Committee public mailing list][steering-public-list]

This notification will include:

* a link to the aforementioned GitHub issue
* the statement providing context on the reason for the vote

There will be a period of two weeks for members of the community to reach
out to Steering Committee members to provide feedback.

Community members may provide feedback by the following methods:

* commenting on the GitHub issue

> * replying to the notification email

Reviewer 6: +1 to Bob's suggestion of removing this line

> * replying to the notification email

Reviewer 21: +1 here from me also.

> * replying to the notification email

Reviewer 3: Removed in https://github.com/kubernetes/steering/commit/e2c68cb9018b8c1ea73ad51cba3018b7c869668c.

* sending an email to the
  [Steering Committee private mailing list][steering-private-list]
* sending a message to individual committee members

After this feedback period, Steering Committee members must vote on the issue
within 48 hours.

If the vote of no confidence is passed, the member in question will be
immediately removed from the committee.

[dev-list]: mailto:dev@kubernetes.io
[steering-private-list]: mailto:steering-private@kubernetes.io
[steering-public-list]: mailto:steering@kubernetes.io
[steering-repo]: https://git.k8s.io/steering

## Voting

In the course of the committee's operations, members will be expected to vote
on decisions within the body's purview.

These votes may be called on agreed-upon platforms by the committee, such as:

* a pull request
* an issue
* a Steering Committee [meeting](#meetings)
* a mailing list

For public business, the vote must be captured on an issue or pull request.

### Routine business

Unless otherwise specified by a process, the requirement for passing a vote is
a **_majority of the [fixed membership of the committee](#composition)_**.

> Example:

Reviewer 6: Another example, which I would suggest we remove from the final text

* 7 (members) / 2 = 3.5
* Round up to the nearest whole number (4)
* 4 members would be required to pass a vote

### Abstention

For any self-elected reason, members of the committee may decide to abstain
from a vote.

Abstaining members will only be considered as contributing to quorum, in the
event that a vote is called in a meeting.

## Meetings

Steering Committee members are generally expected to attend every meeting. We
use the following guidelines to determine whether we have reached quorum and
are able to proceed with a meeting.

### Quorum

Quorum **to meet** is a **_majority of the
[fixed membership of the committee](#composition)_**.

> Example:

Reviewer 6: Another example, suggest removal as per above.

* 7 (members) / 2 = 3.5
* Round up to the nearest whole number (4)
* 4 members in attendance would be required to meet

Quorum **to vote in a meeting** is a **_two-thirds supermajority of the
[fixed membership of the committee](#composition)_**.

> Example:

Reviewer 6: Another example, suggest removal as per above.

Reviewer 21: If/when these are removed, should update the PR description to remove ", with examples" in the Voting section bullet list.

### Vacancies

In the event of a resignation or other loss of an elected committee member, the
next most preferred candidate from the previous election will be offered the
seat.

A maximum of one (1) committee member may be selected this way between
elections.

In case this fails to fill the seat, a special election for that position will
be held as soon as possible.

Examples (non-exhaustive):

> * the next most preferred candidate is no longer interested in sitting on the

Reviewer: currently ... we go down the list.

> * the next most preferred candidate is no longer interested in sitting on the

Reviewer 2: as pointed out in some other comment, this could easily choose the community's least preferred candidate if the field isn't large some year.

> * the next most preferred candidate is no longer interested in sitting on the

Reviewer 3: > This does not say how far down the list we go. If the first non-elected person declines, do we consider the second, third, ... tenth? Or is it only the first runner up who is offered a seat?

@thockin -- Here's some of what was changed in this section:

> This process will continue until the seat is filled.

was removed.

> A maximum of one (1) committee member may be selected this way between
elections.

was added. (It's a few lines up on L85-L86.)

That's meant to say:
Only _the next preferred candidate_ from the previous election will be considered.
If for some reason they can't accept, then we hold a special election.

---

> as pointed out in some other comment, this could easily choose the community's least preferred candidate if the field isn't large some year.

The comment @lavalamp is referencing @jberkus in https://github.com/kubernetes/steering/pull/248#issuecomment-1179276881:

> > 2 down the stack feels arbitrary and doesn't account for the % of votes or skew between one or both of the next 2 and the 3-4 members seated from the election. If you took the top 4 and the skew between the 5th and the 4th is small but the skew between the 4/5th and the 6th is huge... it doesn't feel very representative of the community's will. It makes more sense to go down the stack so long as the candidates are within X% of the last seated candidate otherwise hold a special election.
> > I think these two are potentially sub-optimal outcomes: 1.2..3..4.5.................6 or 1..2.3.4..........5...........6
> 
> I share Kirsten's reservations here. It really depends on the election, and that's hard to incorporate into the rules. For example, last steering election we had 13 candidates, so it's reasonable to assume that anyone in the top 7 is fairly well "liked" by the contributors. BUT ... we've had one steering election where there were 5 candidates for 3 seats, and 2 down means picking the candidate who was the least preferred. If you combine that with the open nature of our elections -- where anyone can run, even if they have no prior association with Kubernetes -- it's a recipe for worry.
> 
> So I'd be more comfortable if using alternates from the prior election ended at 1. That is, the seat is offered to the next-most-preferred candidate. If they refuse it, we have a special election.

> * the next most preferred candidate is no longer interested in sitting on the

Reviewer 2: I think the issue is that "selected" implies asking the person and going down the list until someone says "yes". If you change to "selection attempt" or "at most one person shall be invited this way" then it won't be ambiguous.

> * the next most preferred candidate is no longer interested in sitting on the

Reviewer 3: > suggestion: drop the examples section, as the actual text covers our intent

Done. Resolving convo.

  committee
* the next most preferred candidate cannot sit on the committee, due to some
  committee composition requirement i.e.,
  [maximal representation][maximal-representation]

[Eligible voters][voter-eligibility] from the most recent election will vote in
the special election i.e., eligibility will not be redetermined at the time of
the special election.

A committee member elected in a special election will serve out the remainder
of the term for the person they are replacing, regardless of the length of that
remainder.

[maximal-representation]: /elections.md#maximal-representation
[voter-eligibility]: /elections.md#eligibility-for-voting

### Resignation

If a committee member chooses not to continue in their role, for whatever
self-elected reason, they must notify the committee in writing.

### Removal

#### No confidence

A Steering Committee member may be removed by an affirmative vote of a
**_three-quarters supermajority of the
[fixed membership of the committee](#composition)_**.

Example:

* 7 (members) / 4 = 1.75
* 1.75 * 3 = 5.25
* Round up to the nearest whole number (6)

> * Six (6) affirmative votes would be required to remove a member through a vote

Reviewer 4: Curious: I didn't track all the discussion here.  This is effectively "unanimous".  Are we formulating it this way because "maybe one day the committee will grow to 9 and then 3/4 will be 7/9 (which, as you know, has a certain meaning :)?

Reviewer 10: There was a fair amount of back and forth, but we opted to follow the 3/4 vote required for charter changes

Reviewer: I think we started off using this as an example to illustrate, but it just stuck there. (thinking say we lose a member to resignation and immediately after we end up with a bad situation unrelated to the other person and have to deal with it)

Reviewer 3: > Curious: I didn't track all the discussion here. This is effectively "unanimous". Are we formulating it this way because "maybe one day the committee will grow to 9 and then 3/4 will be 7/9 (which, as you know, has a certain meaning :)?

@thockin --

1. LOL
2. What Bob [said](https://github.com/kubernetes/steering/pull/248#discussion_r917460116):
   > There was a fair amount of back and forth, but we opted to follow the 3/4 vote required for charter changes

   Based on the discussions, folks felt that a two-thirds supermajority was too few.

---

> thinking say we lose a member to resignation and immediately after we end up with a bad situation unrelated to the other person and have to deal with it

@dims -- I think this is a special case that we haven't discussed yet. Let's do so on https://github.com/kubernetes/steering/issues/250.

  of no confidence

The call for a vote of no confidence will happen in a public Steering Committee

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 6: > I'd propose stating that the party in question will be permitted to post (in a comment) a link to a defense/rebuttal on their own platform, and that moderation won't delete it.

I can't get behind this either, as it puts restrictions on our moderation and CoC teams.

What I'm worried about is someone using this process/platform to post a manifesto/screed in their defence. I sincerely hope that this wouldn't be the case, but I also hope that this is a policy we don't need to use in the first place.

*If* the impacted person posts a public comment that doesn't violate any moderation or CoC policies, then it can exist just like any other public comment on the issue. I'm not saying they shouldn't be able to do that. My concern is with elevating or somehow enshrining their "right" to do that as some sort of artifact of the process that cannot be touched.

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 2: An unhinged screed is actually good evidence that removal is correct? I'm not saying we should host it, only that we do people a favor by helping them find it.

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 15: An unhinged screed likely does undue damage to members of our community. I've seen enough of them in communities in the past to not want a repeat. We don't want to platform or direct people to that.

For any removal situation that wouldn't result in a screed, a comment is welcome and more than enough.

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 5: Agreed. We have to consider the case of an SC member being removed specifically for CoC violations that include their posts on community-owned channels. In that case, providing them an additional channel to commit a violation on is counterproductive and could cause harm.  And if we include a right to rebuttal in this process document, we won't be able to remove it.

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 14: > This isn't a trial or a debate.

This looks exactly like a trial with jury comprised of the six other steering members.  Having such a trial without transparent evidence makes it difficult to mount a defense and to have such a trial without the ability to even provide a defense seems even worse.  Claims of "I'm worried about is someone using this process/platform to post a manifesto/screed in their defence" and "providing them an additional channel to commit a violation on is counterproductive and could cause harm" have parallels to previous illegitimate courts such as a those at the Salem witch trials that notably 
1. were created in a rush as a stop-gap ("gave the colony’s legislature the right to establish a court. But the process would take time, and Phips had to act quickly.")
2. made evidence of hearsay ("the admission of rumors, assumptions, and community gossip")
3. prevented challenges to alleged evidence ("most of the witnesses against the accused witches could have been very effectively challenged.")

[source](https://www.history.com/news/salem-witch-trials-justice-legal-legacy#:~:text=The%20second%20key%20legal%20protection,who%20testified%2C%E2%80%9D%20Niehoff%20says.) which concludes these bad examples led to many of guarantees people generally expect from their trials today.  Repeating these errors, for what appears to be similar reasons ("the speedy and vigorous prosecution of such as have rendered themselves obnoxious") is unfortunate.

> meeting and must be documented as a GitHub issue in the committee's

Reviewer: @deads2k please STOP here!

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 19: > I can assure you that the steering committee talks about everything going on with our community constantly, and the other members will be clear on the position of the person impacted.

The statement provided as justification for the vote of no confidence is not primarily for the other steering members, it's for community members to understand why an attempt to remove their elected representative is happening.

The same is true of the response from the impacted individual. It's for community members, not for other steering members, so the community can hear both sides and give well-informed feedback in the time provided.

> This also doesn't restrict the impacted individual from speaking publicly, only that their statement isn't an official artifact of the process.

The impacted individual's statement should *absolutely* be an official artifact of the process. Refusing to put the accusations and the response side-by-side for community members to weigh, and elevating the accusation while limiting the response to a comment or one of many replies is inappropriate.

I don't find the what-if scenarios about violating moderation or CoC policies a compelling reason to demote the response. The same possibility exists with the accusations, and I'd expect both the accusations and the response to be subject to normal moderation. A response from the impacted person that doesn't violate moderation or CoC policies should be placed side-by-side with the accusations.

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 2: I wasn't going to say any more on this, but I guess I'll put in one last thought because I see a different primary concern. I don't care about fairness as an end in itself.

The point is for the community to be able to police the board. That is, decide which faction of the board is actually harming the community--both sides will claim the other side is doing so. This policing function is incompatible with only seeing one side of the argument.

There is already social technology for presenting material some people may experience harm from viewing-- content warnings.

To feel why it is important for the community to have this policing role, you can imagine some issue you care greatly about, and imagine it is a tiny minority on the board.

It is very short-sighted to enact rules which can be used against one's "side" if one's "side" falls out of power.

Rules like these are not about being fair to today's incorrect factions. Rules like these are about allowing future correct factions to survive power grabs. I wish it weren't so, but it is.

> meeting and must be documented as a GitHub issue in the committee's

Reviewer: @lavalamp @liggitt @deads2k thanks for your points. I am with @cblecker here. 

```This also doesn't restrict the impacted individual from speaking publicly, only that their statement isn't an official artifact of the process. We don't need to provide a platform.```

I will support the language as-is currently.

> meeting and must be documented as a GitHub issue in the committee's

Reviewer: @justaugustus @parispittman unless you think otherwise, please mark this thread as resolved.

@lavalamp there are plenty of forums for saying one's piece.

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 3: Given that Steering Committee members are Kubernetes community members, they are covered under what has already been written about the community feedback period:

> There will be a period of two weeks for members of the community to reach
> out to Steering Committee members to provide feedback.
>
> Community members may provide feedback by the following methods:
>
> * commenting on the GitHub issue
> * replying to the notification email
> * sending an email to the
>   [Steering Committee private mailing list][steering-private-list]
> * sending a message to individual committee members

---

> @justaugustus @parispittman unless you think otherwise, please mark this thread as resolved.

Resolving convo.

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 2: > Steering Committee members are Kubernetes community members, they are covered under what has already been written about the community feedback

Sorry, I really wanted to leave this thread, but I can't with that. I am losing the sense that y'all are arguing in good faith. That is like saying the accused at a trial is in the courtroom and can therefore give feedback like anyone else in the courtroom. The accused doesn't need to give feedback to the board, they are *on* the board already. The accused needs to present a case to the **community**, so that the community can judge the board.

I know you don't think this is like a trial, but I believe you are mistaken about that. I sincerely believe that y'all are having a hard time imagining a hypothetical case where this policy is (in your mind) incorrectly aimed at yourself, and if you took 5 minutes and visualized that as hard as you could, you would not be happy with your options.

I don't feel like my objections have actually been understood and responded to. Policies are foreverish, please consider more than just the current situation when writing them.

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 6: @lavalamp I do hear your argument, and I assure you that I am approaching this in extremely good faith.

That said, when weighing the potential harms to the community of providing an enshrined platform to someone that the other elected representatives potentially don't have confidence in to serve in the communities best interest, with the potential benefits you've outlined, unfortunately I don't agree with your viewpoint.

Again, to be explicitly clear, we aren't silencing anyone, including the impacted person. If they make comments on the issue or in other public forums that comply with things like our Code of Conduct, those comments can stand like any other. But that impacted person doesn't have a "right" in my view to say whatever they like, irrespective of harm to the community.

I realize that even after all of this, you may still be dissatisfied with this response. However, I do believe that we have fairly and fulsomely discussed this point, even if that resolution isn't the one you hoped for.

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 2: @justaugustus sure, I'll make a suggestion later today.

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 3: Thanks Daniel!

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 2: That was a less trivial task than expected (as always) -- I made a PR directly against Paris's branch: https://github.com/parispittman/steering/pull/1

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 14: > That was a less trivial task than expected (as always) -- I made a PR directly against Paris's branch: https://github.com/parispittman/steering/pull/1

That PR would alleviate most of my concerns from beginning of this thread.

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 14: > @lavalamp -- Can you present a solution that you feel is more rounded for all of those involved?
> 
> After 355 comments, in lieu of more discussion, I would love to see a solution in the exact copy you'd like us to consider for inclusion in this charter update.

Not starting a new thread, just looking for clarification on an existing thread.

@justaugustus You requested some specific text and it was provided on the same day.  So far I have not seen your comments here or in https://github.com/parispittman/steering/pull/1.  Do you (and the other steering members) plan to review the text you requested and provide feedback before taking a vote on this PR?  So far, only dims has directly commented.

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 6: This is a complex topic, and I'd like to thank you both for your engagement and desire to help the community.

I've read and reviewed your proposal, @lavalamp, multiple times over the course of the last week.

When looking at something like this, in the absence of a win-win proposal that satisfies all concerns, those concerns need to be weighted. I believe strongly, and have a track record of advocating for, transparency in open source. Folks should have, whenever possible, the opportunity to know and understand what is happening in the community. What I'm concerned for and weighing this against, is the possibility that an individual who is impacted by this would use this as a loophole to platform harmful views.

We are looking at extreme scenarios here (and I would absolutely classify the use of this process in the first place as an extreme scenario), and I don't deny there is a possibility of a "rogue board" scenario. That said, I think the likelihood of that scenario (six elected committee members, from at least three different employers engaged in a conspiracy to remove the seventh person) is significantly less likely than the "platforming harm" scenario (all it takes is one committee member to execute).

The other piece here that is weighted into my thought process is that the impacted committee member still has the ability to make statements (as a comment on the issue, a link to their gist as you proposed, on their personal blog, twitter, or favourite podcast) without any additional language. The difference as I see it, is that without the proposed revision, those comments are still subject to the Code of Conduct and the CoCC (not SC) can take action if they are harmful, such as moderate and remove them if they are in a project space.

Nobody in our community, even Steering Committee members, and even ones facing forceful removal from the committee, are "above" the code of conduct in my view.

Again, I appreciate your desire to help the community by providing feedback on this, and I assure you I have taken multiple "beats" to consider this opinion and proposed additional language thoughtfully. However, for the reasons above, I'm -1 to modifying the proposed language to include these changes.

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 2: > The difference as I see it, is that without the proposed revision, those comments are still subject to the Code of Conduct and the CoCC (not SC) can take action if they are harmful, such as moderate and remove them if they are in a project space

I see it the other way around. If we host the artifact it is easy to apply CoC to it? If the person hosts it elsewhere there is little we can do? I didn't intend my proposal to let people post CoC violating subject matter.

> meeting and must be documented as a GitHub issue in the committee's

Reviewer: @lavalamp we take screen shots when we see questionable content go by

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 2: @dims I don't understand what you mean by that

> meeting and must be documented as a GitHub issue in the committee's

Reviewer: @lavalamp typically when we see what we think is a CoC violation somewhere, we take screen shots just in case it gets cleaned up (by design or by accident). I was responding to the " If the person hosts it elsewhere there is little we can do" in your question.

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 2: My point is we can't make them take it down / fix it unless we own the forum.

I continue to think you all are overtraining on the possibility of a CoC violating response. I don't see how that would help the person in question at all.

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 15: @lavalamp those responses rarely help anyone, but they're all too common (just look around at various Big Personalities that have existed in and around linux development communities for decades and the things they've said about queer people/women/disabled people/...).

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 2: In general, sure, but as a response to an attempt to kick someone off a board? That would make very little sense as self defense. I'll change my mind given a few examples of such a thing, if you have them.

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 22: @lavalamp no one here is going to take the bait and attempt to change your mind. That isn't positive nor productive. If this community is important enough to you, you are more than enough capable of researching the topic independently.

> meeting and must be documented as a GitHub issue in the committee's

Reviewer 2: That wasn't disingenuous, but you don't know me and I don't know how to prove that in a short amount of text. Nor do I know how to express the sentiment in a way that sufficiently distinguishes it from people who are attempting to drag out arguments rather than trying to find some objective reason to conclude them, which is how I meant it.

Anyway, it's too late now as people are voting.

[repository][steering-repo].

The call for a vote of no confidence must be made by a current member of the
committee and must be seconded by another current member.

> The committee member who calls for the vote will prepare a short statement

Reviewer 4: do we need to define "short" ?

Reviewer: if we want to pick, it would be 300 words as the bio for elections ( https://github.com/kubernetes/community/search?q=300+words ), we could say approximately 300 words if we insist on defining "short". 

BUT i don't see the necessity to define short here. we are just telling people that it wont be a really long thing i feel.

Reviewer 2: This would only be executed under potentially very antagonistic conditions and IMO "short" needs to be defined. Otherwise there will be a fight about whether the board's actions were valid under the board's own rules.

Author: I think we should remove the word short and leave the rest

Reviewer 3: > I think we should remove the word short and leave the rest

Agreed, for simplicity. Removed in https://github.com/kubernetes/steering/pull/248/commits/d4552c5014a97cf6983c2fd85d50b0bb8eeb0b88.

which provides context on the reason for the vote. This statement must be
seconded by the committee member who seconded the vote.

Once a vote of no confidence has been called, the committee will notify the
community through the following channels:

* the [community mailing list][dev-list]
* the [Steering Committee public mailing list][steering-public-list]

This notification will include:

* a link to the aforementioned GitHub issue
* the statement providing context on the reason for the vote

There will be a period of two weeks for members of the community to reach
out to Steering Committee members to provide feedback.

Community members may provide feedback by the following methods:

* commenting on the GitHub issue
* replying to the notification email
* sending an email to the
  [Steering Committee private mailing list][steering-private-list]
* sending a message to individual committee members

After this feedback period, Steering Committee members must vote on the issue

> within 48 hours.

Reviewer 2: I would state, for extra clarity, what happens if this passes or fails, right now it's implicit. Probably most important is to state when it is effective if it passes.

Reviewer 3: Added in https://github.com/kubernetes/steering/pull/248/commits/a2e2434172a7e57ec6dd727411a1c88b06e2b91c.

meeting and must be documented as a GitHub issue in the committee's
[repository][steering-repo].

The call for a vote of no confidence must be made by a current member of the
committee and must be seconded by another current member.

The committee member who calls for the vote will prepare a statement which
provides context on the reason for the vote. This statement must be seconded by
the committee member who seconded the vote.

Once a vote of no confidence has been called, the committee will notify the
community through the following channels:

* the [community mailing list][dev-list]
* the [Steering Committee public mailing list][steering-public-list]

This notification will include:

* a link to the aforementioned GitHub issue
* the statement providing context on the reason for the vote

There will be a period of two weeks for members of the community to reach
out to Steering Committee members to provide feedback.

Community members may provide feedback by the following methods:

* commenting on the GitHub issue
* replying to the notification email
* sending an email to the
  [Steering Committee private mailing list][steering-private-list]

> * sending a message to individual committee members

Reviewer 23: What should happen within the SC if someone picks this route? Or should folks make no assumption about folks forwarding the message on internally etc.

[voter-eligibility]: /elections.md#eligibility-for-voting

### Resignation

If a committee member chooses not to continue in their role, for whatever
self-elected reason, they must notify the committee in writing.

### Removal

#### No confidence

A Steering Committee member may be removed by an affirmative vote of a
**_three-quarters supermajority of the
[fixed membership of the committee](#composition)_**.

Example:

* 7 (members) / 4 = 1.75
* 1.75 * 3 = 5.25
* Round up to the nearest whole number (6)
* Six (6) affirmative votes would be required to remove a member through a vote
  of no confidence

The call for a vote of no confidence will happen in a public Steering Committee meeting and a
GitHub issue in kubernetes/steering by a current member and must be seconded by
another current member.

Once a vote of no confidence has been called, the committee will notify the
community through the following channels:

* the [community mailing list][dev-list]
* the [Steering Committee public mailing list][steering-public-list]

This notification will include:

* a link to the aforementioned GitHub issue

> * a short statement from non-affected, seated committee members providing

Reviewer 6: I would suggest this statement be agreed to by the member making the motion, and the seconder. This way it represents more than one voice, but at the same time, doesn't require all six folks to bikeshed on the statement prior to a vote being called. The remaining members would agree to this statement when they vote by voting in favour of it.

Reviewer: sounds good

Reviewer 3: Fixed in https://github.com/kubernetes/steering/pull/248/commits/4e46d388f05fd34c6c2527fa402666cb55913dc7.

  context on the reason for the vote

There will be a period of two weeks for members of the community to reach
out to Steering Committee members to provide feedback, which can include

> non-binding votes.

Reviewer 6: All for feedback (privately or publically), but I would suggest removal of the non-binding votes piece, as this could quickly turn into brigading. I would say that +1s or -1s with no other content aren't particularly valuable in this particular case.

Reviewer: good point. agree.

Reviewer 2: agree with removing the last clause completely. This sentence has gotten scrambled anyway, at the moment it appears to say community members can make non-binding votes, which makes no sense.

Reviewer 3: Fixed in https://github.com/kubernetes/steering/pull/248/commits/c12de21656423b8840575b1254591eb293b58715.

[voter-eligibility]: /elections.md#eligibility-for-voting

### Resignation

If a committee member chooses not to continue in their role, for whatever

> self-elected reason, they must notify the committee in writing.

Reviewer 24: Do we want to specify what "in writing" amounts to? An email to steering-private@ ?

Reviewer 5: IMHO, I don't think that we do.

Reviewer 3: > IMHO, I don't think that we do.

(Agreed. Resolving...)

### Removal

#### No confidence

A Steering Committee member may be removed by an affirmative vote of a
**_three-quarters supermajority of the
[fixed membership of the committee](#composition)_**.

> Example:

Reviewer: Can we say "Currently this translates to" instead of "Example"?

Reviewer 3: Leaving as-is, as this section matches the other vote `Example`s elsewhere in the charter.

Reviewer 3: ...additionally, I added the `Composition`, so someone could derive the correct numbers based on the fixed membership of the committee.

The examples just happen to be based on current fixed composition of the committee.
Should the composition of the committee change in the future, any instances of "currently" would have to be corrected.

* 7 (members) / 4 = 1.75
* 1.75 * 3 = 5.25
* Round up to the nearest whole number (6)

> * Six (6) affirmative votes would be required to remove a member through a vote

Reviewer: Convert the last bullet into a sentence starting with "So this translates to Six"

Reviewer 3: Leaving as-is, as this section matches the other vote `Example`s elsewhere in the charter.

  of no confidence

The call for a vote of no confidence will happen in a public Steering Committee meeting and a

> GitHub issue in kubernetes/steering by a current member and must be seconded by

Reviewer: "kubernetes/steering" -> "kubernetes/steering repository"

Reviewer 24: perhaps also make "kubernetes/steering repository" a self-link to the repo here for clarity that doesn't lengthen the rendered document.

Reviewer 3: Fixed in https://github.com/kubernetes/steering/pull/248/commits/107673f27720e863a7c2f0c460eafe3f98a7e2d6.
